File: domain/src/main/java/mixter/domain/identity/SessionId.java
package mixter.domain.identity;

import java.util.UUID;

public class SessionId {
    private String value;

    public SessionId(String value) {
        if (value.trim().isEmpty()) {
            throw new SessionIdCannotBeEmpty();
        }
        this.value = value;
    }

    public static SessionId generate() {
        return new SessionId(UUID.randomUUID().toString());
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        SessionId sessionId = (SessionId) o;

        return !(value != null ? !value.equals(sessionId.value) : sessionId.value != null);

    }

    @Override
    public int hashCode() {
        return value != null ? value.hashCode() : 0;
    }

    @Override
    public String toString() {
        return value;
    }
}

File: Mixter.Domain/Identity/SessionId.cs
using System;

namespace Mixter.Domain.Identity
{
    public struct SessionId
    {
        public string Id { get; private set; }

        public SessionId(string id)
            : this()
        {
            Id = id;
        }

        public static SessionId Generate()
        {
            return new SessionId(Guid.NewGuid().ToString());
        }
    }
}
